src/components/Homepage/carousel.jsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import { Carousel } from 'react-responsive-carousel';
+import 'react-responsive-carousel/lib/styles/carousel.min.css';
+import axios from 'axios';
+import Car1 from '../../images/audi-png-45298.png';
+import Car2 from '../../images/audi-png-45306.png';
+import Car3 from '../../images/audi-png-45324.png';
+import Heading from './mainheading';
+import IconsCar from './icons_car';
+
+const MyCarousel = () => {
+  const [carData, setCarData] = useState([]);
+  const [randomCars, setRandomCars] = useState([]);
+
+  useEffect(() => {
+    const fetchCarData = async () => {
+      try {
+        const response = await axios.get('http://localhost:4000/api/v1/items');
+        setCarData(response.data);
+        setRandomCars(generateRandomCars(response.data.length));
+      } catch (error) {
+        console.error('Error fetching car data:', error);
+      }
+    };
+
+    fetchCarData();
+  }, []);
+
+  const generateRandomCars = (count) => {
+    const carImages = [Car1, Car2, Car3];
+
+    return Array.from({ length: count }, () => {
+      const randomIndex = Math.floor(Math.random() * carImages.length);
+      return carImages[randomIndex];
+    });
+  };
+
+  const handleReservation = (carId) => {
+    localStorage.setItem('selectedCarId', carId);
+    // window.location.href = `/items/${carId}`;
```

Comment:
- [ ] There is still 1 piece of commented code on this file.  Kindly remove it to preserve the repo's cleanliness. 👍